src/components/learner-credit-management/assignment-modal/NewAssignmentModalDropdown.jsx

```diff
@@ -59,26 +60,31 @@ const NewAssignmentModalDropdown = ({
         <Dropdown.Header className="text-uppercase">
           {courseRuns.length > 0 ? intl.formatMessage(messages.byDate) : intl.formatMessage(messages.noAvailableDates) }
         </Dropdown.Header>
-        {courseRuns.length > 0 && courseRuns.map(courseRun => (
-          <Dropdown.Item
-            key={courseRun.key}
-            data-courserunkey={courseRun.key}
-            onMouseDown={() => setClickedDropdownItem(courseRun)}
-            onMouseUp={() => setClickedDropdownItem(null)}
-          >
-            <Stack>
-              {intl.formatMessage(messages.startDate, {
-                startLabel: startLabel(courseRun),
-                startDate: dayjs(courseRun.start).format(SHORT_MONTH_DATE_FORMAT),
-              })}
-              <span className={classNames('small', { 'text-muted': getDropdownItemClassName(courseRun) })}>
-                {intl.formatMessage(messages.enrollBy, {
-                  enrollByDate: dayjs(courseRun.enrollBy).format(SHORT_MONTH_DATE_FORMAT),
+        {courseRuns.length > 0 && courseRuns.map(courseRun => {
+          if (isLoading) {
```

Comment:
[clarification] When the async loading is resolved, will we still have the same number of `courseRuns` we're iterating over in this code path? E.g., if the loading completes, and we determine a restricted run is NOT contained by the catalog, does the `courseRuns` list end up having 1 less run to iterate through?

If so, I wonder if it's worth keeping the `Skeleton` loading dropdown items separate from the iteration of `courseRuns.map(...)`?

For example, when `isLoading: true`, it could make sense to just default to showing ~3 skeleton dropdown items before showing all available runs once loading is complete so as not to show a loading state for a course run that would ultimately end up getting removed anyway?

Reply:
The way I designed it: the default number of skeleton options to show in the dropdown is always just the number of confirmed assignable runs at any given time. Before any API calls to contains_content_items resolves, the number of skeletons should be equal to the number of unrestricted runs, then as each API call for each restricted run resolves, the number of skeletons will grow (or not) until finally isLoading flips to false.  At that point, a final re-render of the component should reveal the assignable runs.

While the explanation above is more complex than just "default to 3 skeletons", the nice thing is that there's no extra code to handle the extra case.

Reply:
Thanks for the explanation/clarification. Approved with the nit considerations :)